### PR TITLE
falcon-i18n translations

### PR DESCRIPTION
**Pull request by tutok**
* [x] remove react-i18next, removed namespace 
* [x] add T and I18n components
* [x] language changing
* [ ] translations for checkout
* [x] translations for new components on master
* [x] translations for `FormField` component
* [x] documentation - will be added in separate PR as we have another repo for docs

### CODE REVIEW

File: packages/falcon-ecommerce-uikit/src/Footer/Footer.tsx [hunk 1]
```diff
@@ -70,24 +71,29 @@ const languageSectionTheme: DefaultThemeProps = {
     }
   }
 };
-export const Footer: React.SFC<FooterData & { translations: FooterTranslations }> = ({
+
+export const Copyright = () => (
+  <>
+    <T id="copyright" /> {new Date().getFullYear()}
```
Comment (Borales) [inline packages/falcon-ecommerce-uikit/src/Footer/Footer.tsx]: @tutok just a minor thing, why can't we pass `year` to the translation and handle its place there? For example, github has `© 2018 GitHub, Inc.`

File: packages/falcon-ecommerce-uikit/src/Footer/LanguageSwitcher.tsx [hunk 2]
```diff
@@ -2,12 +2,23 @@ import React from 'react';
 import { Dropdown, DropdownLabel, DropdownMenu, DropdownMenuItem } from '@deity/falcon-ui';
 import { LanguageItem } from './FooterQuery';
 
+const addCIModeLanguage = (languages: LanguageItem[]) => {
+  if (process.env.NODE_ENV === 'development') {
+    if (!languages.find(x => x.code === 'cimode')) {
+      languages.unshift({ code: 'cimode', name: 'CI mode', active: false });
```
Comment (Borales) [inline packages/falcon-ecommerce-uikit/src/Footer/LanguageSwitcher.tsx]: @tutok what does this `CI mode` mean?

Comment (tutok) [reply packages/falcon-ecommerce-uikit/src/Footer/LanguageSwitcher.tsx]: [cimode](https://www.i18next.com/overview/api#changelanguage) is quite useful when you would like to check what is handled via i18next, it helps also to find which transaltion `key` is used for specific text.
image

File: yarn.lock [hunk 3]
```diff
@@ -8563,11 +8533,21 @@ i18next-xhr-backend@^1.5.1:
...
+  version "12.0.0"
+  resolved "https://registry.npmjs.org/i18next/-/i18next-12.0.0.tgz#27c1494219dde0451a8d714d5bfc19bf055d51bb"
+  integrity sha512-Zy/nFpmBZxgmi6k9HkHbf+MwvAwiY5BDzNjNfvyLPKyalc2YBwwZtblESDlTKLDO8XSv23qYRY2uZcADDlRSjQ==
+
 i18next@^11.6.0:
   version "11.10.2"
   resolved "https://registry.npmjs.org/i18next/-/i18next-11.10.2.tgz#e5f10346f6320ecf15595419926c25255381a56c"
   integrity sha512-1rowdX8PqrvsdFhYb3v0A/LlIHLQL1HTa4ia29IzhvNAg2fesNV7R1jXibWLmLQdz3FfTB8RuqSqDEjIawXruA==
 
+i18next@^12.0.0:
```
Comment (Borales) [inline yarn.lock]: @tutok you have 2 `i18next` versions, should we pick 1?

File: yarn.lock [hunk 4]
```diff
@@ -16036,6 +16022,11 @@ typescript-estree@5.3.0:
     lodash.unescape "4.0.1"
     semver "5.5.0"
 
+typescript@3.1.1:
+  version "3.1.1"
+  resolved "https://registry.npmjs.org/typescript/-/typescript-3.1.1.tgz#3362ba9dd1e482ebb2355b02dfe8bcd19a2c7c96"
+  integrity sha512-Veu0w4dTc/9wlWNf2jeRInNodKlcdLgemvPsrNpfu5Pq39sgfFjvIIgTsvUHCoLBnMhPoUA+tFxsXjU6VexVRQ==
+
 typescript@3.2.1:
```
Comment (Borales) [inline yarn.lock]: @tutok 2 diff versions of Typescript

File: packages/falcon-ecommerce-uikit/src/Product/Product.tsx [hunk 5]
```diff
@@ -169,64 +158,73 @@ export class Product extends React.PureComponent<{ product: any; translations: P
...
+                },
+                formik: { values, errors, setFieldValue, submitCount },
+                productConfigurator
+              }: any) => (
+                <ProductDetailsLayout>
+                  <FlexLayout gridArea={Area.gallery} alignItems="center" justifyContent="center">
+                    <ProductGallery items={product.gallery} />
+                  </FlexLayout>
+                  <Text fontSize="sm" gridArea={Area.sku}>
+                    {`${t('product.sku')}: ${product.sku}`}
```
[the remark this answers is not in the document]
Comment (tutok) [reply packages/falcon-ecommerce-uikit/src/Product/Product.tsx]: in my opinion moving only `:` into `translations.json` is not good. a better idea will update `product.sku`  key from `"SKU"` into `"SKU: {{sku}}"`

File: packages/falcon-i18n/package.json [hunk 6]
```diff
@@ -3,8 +3,56 @@
...
+    "i18next": "^12.0.0",
+    "prop-types": "^15.6.2",
+    "react": "~16.4.2",
+    "react-dom": "~16.4.2",
+    "rimraf": "2.6.2",
+    "rollup": "0.66.4",
+    "rollup-plugin-babel": "4.0.3",
+    "rollup-plugin-commonjs": "9.1.8",
+    "rollup-plugin-node-resolve": "3.4.0",
+    "typescript": "3.1.1"
```
[the remark this answers is not in the document]
Comment (tutok) [reply packages/falcon-i18n/package.json]: ok maybe I merged something wrong, will look at this. thanks

File: packages/falcon-ecommerce-uikit/package.json [hunk 7]
```diff
@@ -73,15 +74,16 @@
     "@babel/preset-env": "7.1.6",
     "@babel/preset-react": "7.0.0",
     "@babel/preset-typescript": "7.1.0",
+    "@deity/falcon-i18n": "0.1.0",
     "@deity/falcon-server": "^0.1.0",
     "@deity/falcon-shop-extension": "^0.1.0",
     "@deity/falcon-ui": "^0.1.0",
     "@types/enzyme": "^3.1.15",
     "@types/enzyme-adapter-react-16": "^1.0.3",
+    "@types/i18next": "11.9.3",
```
[the remark this answers is not in the document]
Comment (Borales) [reply packages/falcon-ecommerce-uikit/package.json]: @tutok I was referencing to `@types/i18next: 12.1.0`, i18next itself - haven't checked

[the remark this answers is not in the document]
Comment (tutok) [reply packages/falcon-ecommerce-uikit/package.json]: done

File: yarn.lock (hunk 4 above)
Comment (tutok) [reply yarn.lock]: done

File: yarn.lock (hunk 3 above)
Comment (tutok) [reply yarn.lock]: done

File: packages/falcon-ecommerce-uikit/src/Checkout/CheckoutLogic.tsx [hunk 8]
```diff
@@ -153,7 +153,7 @@ class CheckoutLogicImpl extends React.Component<CheckoutLogicProps, CheckoutLogi
     this.setLoading(true, () => {
       // trigger mutationt that will return available shipping options
       this.props.client
-        .mutate({
+        .mutate<any>({
```
[the remark this answers is not in the document]
Comment (tutok) [reply packages/falcon-ecommerce-uikit/src/Checkout/CheckoutLogic.tsx]: done

**Comment by tutok** [comment]
resolves #175

**Pull request merged**